adaptors/dummy/test/e2e/usability/usability_test.go

```diff
@@ -0,0 +1,552 @@
+
+	BeforeEach(func() {
+		// create device link
+		deployDummyDeviceLink()
+	})
+
+	AfterEach(func() {
+		// delete device link, ignore error
+		_ = k8sCli.DeleteAllOf(testCtx, &edgev1alpha1.DeviceLink{}, client.InNamespace(deviceLink.Namespace))
+	})
+
+	Context("edit dummy device link spec", func() {
+		Specify("if invalid node spec", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when invalid node spec", invalidNodeSpec)
+
+			By("then the device link status should be NodeExisted false", isNodeExistedFalse)
+
+			By("when correct node spec", correctNodeSpec)
+
+			By("then the device link status should be NodeExisted true", isNodeExistedTrue)
+
+		})
+
+		Specify("if invalid model spec", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when invalid model spec", invalidModelSpec)
+
+			By("then the device link status should be ModelExisted false", isModelExistedFalse)
+
+			By("when correct model spec", correctModelSpec)
+
+			By("then the device link status should be ModelExisted true", isModelExistedTrue)
+
+		})
+
+		Specify("if invalid adaptor spec", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when invalid adaptor spec", invalidAdaptorSpec)
+
+			By("then the device link status should be AdaptorExisted false", isAdaptorExistedFalse)
+
+			By("when correct adaptor spec", correctAdaptorSpec)
+
+			By("then the device link status should be AdaptorExisted true", isAdaptorExistedTrue)
+
+		})
+
+		Specify("if invalid device spec", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when invalid device spec", invalidDeviceSpec)
+
+			By("then the device link status should be DeviceConnected false", isDeviceConnectedFalse)
+
+			By("when correct device spec", correctDeviceSpec)
+
+			By("then the device link status should be DeviceConnected true", isDeviceConnectedTrue)
+
+		})
+
+		Specify("if invalid device spec to another valid spec", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when edit spec.template.spec.gear to fast", editGearToFast)
+
+			By("then the device link work in config(fast)", isDeviceConfigFast)
+
+		})
+	})
+
+	Context("restart adaptors pods", func() {
+		Specify("if delete dummy adaptor pods", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when delete dummy adaptor pods", deleteDummyAdaptorPods)
+
+			By("then the device link status should be AdaptorExisted false", isAdaptorExistedFalse)
+
+		})
+
+		// this specify does not pass currently
+		XSpecify("if delete octopus limbs pods", func() {
+
+			By("given the device link is deployed", isDummyDeviceLinkDeployed)
+
+			By("when delete octopus limbs pod", deleteLimbsPods)
+
+			By("then the dummy adaptor pods become error", isDummyAdaptorPodsError)
+
+		})
+	})
+
+	Context("delete DummySpecialDevice CRD", func() {
```

Comment:
One `Specify` doesn't need a `Context`, itself was a context.

Reply:
Use `delete dummy device model` as the description instead of `delete DummySpecialDevice CRD`, although we all know here is deleting the CRD.

Reply:
updated